packages/sdk/test/cross-chain-messenger.spec.ts

```diff
@@ -25,6 +25,22 @@ describe('CrossChainMessenger', () => {
   let l2Signer: any
   before(async () => {
     ;[l1Signer, l2Signer] = await ethers.getSigners()
+    // ethers.getSigners() is out of date - recent versions of geth
+    // include a balance check as part of the estimateGas() call.
+    for (const signer of [l1Signer, l2Signer]) {
+      const estimateGas = signer.provider.estimateGas
+      signer.provider.estimateGas = async function (tx) {
```

Comment:
So if I'm understanding this correctly, the issue here is geth reverts on balance checks. What I'm failing to understand though is we already have these tests running against geth via our devnet tests and against anvil via our anvil/vitest tests

Reply:
https://github.com/ethereum/go-ethereum/blob/dfb3d46098520e90811c6ada3f8e142789c25832/eth/gasestimator/gasestimator.go#L79C1-L80C1 think new versions of geth do fail the gas estimate if the balance is lower than the call value. I'll update the test to check for strict inequality since a zero balance is permitted for non-payable functions.

Reply:
for example, cast rpc eth_estimateGas '{"to":"0x00a149B262EA8ACA092Af1c43a899e7B18C5A4De", "from":"0x00a149B262EA8ACA092Af1c43a899e7B18C5A4De","value":"0x1"}' fails on most networks due to insufficient balance while cast rpc eth_estimateGas '{"to":"0x00a149B262EA8ACA092Af1c43a899e7B18C5A4De", "from":"0x00a149B262EA8ACA092Af1c43a899e7B18C5A4De","value":"0x0"}' succeeds.

Reply:
hmmm definitely confusing to me because I'm pretty sure we are using a new version of geth for our devnet tests. I think you are right though and I just don't know enough to be able to approve this pr. @smartcontracts will check it out within a day or 2. Thank you for patience.